### PR TITLE
[arrow] Add static element converter

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Sometimes, we need to convert a single element to arrow field vector

### CODE REVIEW

File: paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.paimon.arrow.writer.ArrowFieldWriterFactoryVisitor;
+import org.apache.paimon.data.InternalRow;
+import org.apache.paimon.types.DataField;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.FieldVector;
+
+import static org.apache.paimon.arrow.ArrowUtils.createVector;
+
+/** Convert a static value which contained in InternalRow to a FieldVector. */
```
Comment (yuzelin) [inline paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java:31]: Pass the object value and build GenericRow internal?

**Review by yuzelin — APPROVED** [review]
+1

File: paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/** Convert a static value to a FieldVector. */
+public class StaticElementConverter {
+
+    private final GenericRow row;
+    private final FieldVector fieldVector;
+    private final ArrowFieldWriter writer;
+
+    private int pos = 0;
+
+    public StaticElementConverter(
```
Comment (JingsongLi) [inline paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java:40]: Why this is a class? It seems a static method is OK.

Comment (JingsongLi) [reply paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java:40]: We can generate a max size static vector?

Comment (leaves12138) [reply paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java:40]: We can't fill the max size static vector, it too waste of memory

Comment (leaves12138) [reply paimon-arrow/src/main/java/org/apache/paimon/arrow/vector/StaticElementConverter.java:40]: It's better to fill it when needed

**Pull request merged**